app/lib/actions/katello/host/attach_subscriptions.rb

```diff
@@ -8,12 +8,16 @@ def queue
 
         def plan(host, pools_with_quantities_params)
           action_subject(host)
+
+          subscription_facet = host.subscription_facet
+          fail _("Please register host '%s' before attaching subscriptions") % host.name unless subscription_facet
```

Comment:
Functionality works great, but looking at the wording again - let's remove the word 'please'. Sorry for suggesting it :)